ProcessLib/TwoPhaseFlowWithPrho/TwoPhaseFlowWithPrhoMaterialProperties.h

```diff
@@ -0,0 +1,270 @@
+    */
+    double calculatedSwdX(double const pl, const double /*X*/, const double S,
+                          const double rho_wet_h2, double const T,
+                          int current_material_id) const
+    {
+        const double pg = pl +
+                          _capillary_pressure_models[current_material_id]
+                              ->getCapillaryPressure(S);
+        double const rho_equilibrium_wet_h2 = pg * Henry_const * molar_mass_h2;
+        if ((1 - S) < (rho_equilibrium_wet_h2 - rho_wet_h2))
+        {
+            return 0.0;
+        }
+        double const drhoh2wet_dpg = Henry_const * molar_mass_h2;
+        double const drhoh2nonwet_dpg = molar_mass_h2 / R / T;
+        double const alpha =
+            ((drhoh2nonwet_dpg - drhoh2wet_dpg) * (1 - S) + drhoh2wet_dpg);
+        double const beta =
+            (drhoh2nonwet_dpg - drhoh2wet_dpg) *
+            pg;  // NOTE here should be PG^h, but we ignore vapor
+        double const dPC_dSw =
+            _capillary_pressure_models[current_material_id]->getdPcdS(S);
+        return -1 / (beta - alpha * dPC_dSw);
+    }
+    /**
+    * Calculate the derivatives using the analytical way
+    */
+    double calculatedXmdX(double pl, double Sw, double rho_wet_h2, double dSwdX,
+                          int current_material_id) const
+    {
+        const double pg = pl +
+                          _capillary_pressure_models[current_material_id]
+                              ->getCapillaryPressure(Sw);
+        double const rho_equilibrium_wet_h2 = pg * Henry_const * molar_mass_h2;
+        double const dPC_dSw =
+            _capillary_pressure_models[current_material_id]->getdPcdS(Sw);
+        if ((1 - Sw) < (rho_equilibrium_wet_h2 - rho_wet_h2))
+            return 1.0;
+        return Henry_const * molar_mass_h2 * dPC_dSw * dSwdX;
+    }
+    /**
+    * Calculate the derivatives using the analytical way
+    */
+    double calculatedXmdP(double pl, double Sw, double rho_wet_h2, double dSwdP,
+                          int current_material_id) const
+    {
+        const double pg = pl +
+                          _capillary_pressure_models[current_material_id]
+                              ->getCapillaryPressure(Sw);
+        double const rho_equilibrium_wet_h2 = pg * Henry_const * molar_mass_h2;
+        double const dPC_dSw =
+            _capillary_pressure_models[current_material_id]->getdPcdS(Sw);
+        if ((1 - Sw) < (rho_equilibrium_wet_h2 - rho_wet_h2))
+            return 0.0;
+        return Henry_const * molar_mass_h2 * (1 + dPC_dSw * dSwdP);
+    }
+
+private:
+    /**Henry constant, Please refer to
+    * De Nevers N. Physical and chemical equilibrium for chemical engineers[M].
+    * John Wiley & Sons, 2012.
+    */
+    double const Henry_const = 7.65e-6;  // mol/Pa./m3
+    double const& molar_mass_h2 = MaterialLib::PhysicalConstant::MolarMass::H2;
+    double const& R = MaterialLib::PhysicalConstant::IdealGasConstant;
```

Comment:
@wenqing I think one can write
```c++
using MaterialLib::PhysicalConstant::IdealGasConstant;
```
and then the IdealGasConstant is available in the current scope.